src/lib.rs

```diff
@@ -964,6 +964,22 @@ impl std::convert::TryInto<UdpSocket> for Client {
     }
 }
 
+impl std::clone::Clone for Client {
+    fn clone(&self) -> Self {
+        Client {
+            socket: match &self.socket {
+                SocketType::Udp(socket) => SocketType::Udp(socket.try_clone().unwrap()),
```

Comment:
I am skeptical that implementing `Clone` is a good idea - most developers assume that clone should never panic. If you really need the functionality, having a `try_clone` just like the standard lib does would be a better choice.